### PR TITLE
Enable IA3 for Gemma and Starcoder

### PR DESCRIPTION
**Pull request by cuichenx**
# What does this PR do ?

Add IA3 support for models that specify `kv_channels`, `num_query_groups` and have `bias`, e.g. Gemma, Starcoder, and Llama 70B

**Collection**: NLP

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by cuichenx** [comment]
jenkins

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!

File: nemo/collections/nlp/modules/common/megatron/adapters/mcore_mixins.py [hunk 1]
```diff
@@ -256,9 +256,11 @@ def mcore_register_adapters(self):
 
     def forward(self, hidden_states):
         # [s, b, 4 * h/p]
-        linear_fc1_output, bias_parallel = self.linear_fc1(hidden_states)
-
-        intermediate_parallel, layernorm_output = linear_fc1_output
+        if self.linear_fc1.te_return_bias:
```
Comment (yaoyu-33) [inline nemo/collections/nlp/modules/common/megatron/adapters/mcore_mixins.py:259]: no longer support non mcore path?

Comment (cuichenx) [reply nemo/collections/nlp/modules/common/megatron/adapters/mcore_mixins.py:259]: this file `mcore_mixins.py` is only for mcore models

**Pull request merged**